aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -94,6 +94,31 @@ void div_kernel(TensorIterator& iter) {
   }
 }
 
+void remainder_kernel(TensorIterator& iter) {
+  if (isIntegralType(iter.dtype(), /*includeBool*/ false)) {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "remainder_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a, scalar_t b) -> scalar_t {
+        TORCH_CHECK(b != 0, "ZeroDivisionError");
```

Comment:
I'm a bit confused about what is going on here.

In my nightly version of PyTorch, the following happens when I have a tensor with remainder 0:
On CPU, I get Floating point exception (core dumped)
On CUDA, I get a tensor filled with 4294967295.

Is that consistent with what you are seeing?  Why the change to make (at least) CUDA an error?

Reply:
Yeah I do see that behavior. Shouldn't we change that though? It seems better to have a catchable exception rather than a core dump. And if the CPU version raises an exception, shouldn't the CUDA version as well?

Reply:
Changing the crash into an exception for `div` was considered high pri: https://github.com/pytorch/pytorch/issues/327

I suspect that changing this behaviour for `remainder` (and also `fmod`) is also desirable.